GitHub authentication failures updating statuses after upgrading to 1.24

After upgrading to 1.24, the plugin is no longer able to update commit statuses, though the "Connect to API" button in the global settings successfully connects to GitHub.

From the console output:
```
Using context: Jenkins builder
FileNotFoundException means that the credentials Jenkins is using is probably wrong. Or the user account does not have write access to the repo.
java.io.FileNotFoundException: {"message":"Not Found","documentation_url":"https://developer.github.com/v3"}
	at org.kohsuke.github.Requester.handleApiError(Requester.java:494)
	at org.kohsuke.github.Requester._to(Requester.java:245)
	at org.kohsuke.github.Requester.to(Requester.java:191)
	at org.kohsuke.github.GHRepository.createCommitStatus(GHRepository.java:780)
	at org.jenkinsci.plugins.ghprb.extensions.status.GhprbSimpleStatus.createCommitStatus(GhprbSimpleStatus.java:175)
	at org.jenkinsci.plugins.ghprb.extensions.status.GhprbSimpleStatus.onBuildStart(GhprbSimpleStatus.java:124)
	at org.jenkinsci.plugins.ghprb.GhprbBuilds.onStarted(GhprbBuilds.java:108)
	at org.jenkinsci.plugins.ghprb.GhprbBuildListener.onStarted(GhprbBuildListener.java:19)
	at org.jenkinsci.plugins.ghprb.GhprbBuildListener.onStarted(GhprbBuildListener.java:12)
	at hudson.model.listeners.RunListener.fireStarted(RunListener.java:215)
	at hudson.model.Run.execute(Run.java:1740)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
	at hudson.model.ResourceController.execute(ResourceController.java:98)
	at hudson.model.Executor.run(Executor.java:374)
Caused by: java.io.FileNotFoundException: https://api.github.com/repos/k8s-bot/kubernetes/statuses/e50c01f6bfc4c6859aa7de9e76aa4ead9fe80558
	at sun.reflect.GeneratedConstructorAccessor148.newInstance(Unknown Source)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:526)
	at sun.net.www.protocol.http.HttpURLConnection$6.run(HttpURLConnection.java:1676)
	at sun.net.www.protocol.http.HttpURLConnection$6.run(HttpURLConnection.java:1674)
	at java.security.AccessController.doPrivileged(Native Method)
	at sun.net.www.protocol.http.HttpURLConnection.getChainedException(HttpURLConnection.java:1672)
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1245)
	at sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:254)
	at org.kohsuke.github.Requester.parse(Requester.java:451)
	at org.kohsuke.github.Requester._to(Requester.java:224)
	... 12 more
Caused by: java.io.FileNotFoundException: https://api.github.com/repos/k8s-bot/kubernetes/statuses/e50c01f6bfc4c6859aa7de9e76aa4ead9fe80558
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1625)
	at java.net.HttpURLConnection.getResponseCode(HttpURLConnection.java:468)
	at sun.net.www.protocol.https.HttpsURLConnectionImpl.getResponseCode(HttpsURLConnectionImpl.java:338)
	at org.kohsuke.github.Requester.parse(Requester.java:447)
	... 13 more
```

Thanks for pointing this out! It looks that credentials were reset during update. 
But first problem is that webhook github gets 500 error when PR is made:
```
<!DOCTYPE html><html><head resURL="/static/7af60c76">
    

    <title>Jenkins [Jenkins]</title><link rel="stylesheet" href="/static/7af60c76/css/style.css" type="text/css" /><link rel="stylesheet" href="/static/7af60c76/css/color.css" type="text/css" /><link rel="stylesheet" href="/static/7af60c76/css/responsive-grid.css" type="text/css" /><link rel="shortcut icon" href="/static/7af60c76/favicon.ico" type="image/vnd.microsoft.icon" /><script>var isRunAsTest=false; var rootURL=""; var resURL="/static/7af60c76";</script><script src="/static/7af60c76/scripts/prototype.js" type="text/javascript"></script><script src="/static/7af60c76/scripts/behavior.js" type="text/javascript"></script><script src='/adjuncts/7af60c76/org/kohsuke/stapler/bind.js' type='text/javascript'></script><script src="/static/7af60c76/scripts/yui/yahoo/yahoo-min.js"></script><script src="/static/7af60c76/scripts/yui/dom/dom-min.js"></script><script src="/static/7af60c76/scripts/yui/event/event-min.js"></script><script src="/static/7af60c76/scripts/yui/animation/animation-min.js"></script><script src="/static/7af60c76/scripts/yui/dragdrop/dragdrop-min.js"></script><script src="/static/7af60c76/scripts/yui/container/container-min.js"></script><script src="/static/7af60c76/scripts/yui/connection/connection-min.js"></script><script src="/static/7af60c76/scripts/yui/datasource/datasource-min.js"></script><script src="/static/7af60c76/scripts/yui/autocomplete/autocomplete-min.js"></script><script src="/static/7af60c76/scripts/yui/menu/menu-min.js"></script><script src="/static/7af60c76/scripts/yui/element/element-min.js"></script><script src="/static/7af60c76/scripts/yui/button/button-min.js"></script><script src="/static/7af60c76/scripts/yui/storage/storage-min.js"></script><script src="/static/7af60c76/scripts/hudson-behavior.js" type="text/javascript"></script><script src="/static/7af60c76/scripts/sortable.js" type="text/javascript"></script><script>crumb.init("", "");</script><link rel="stylesheet" href="/static/7af60c76/scripts/yui/container/assets/container.css" type="text/css" /><link rel="stylesheet" href="/static/7af60c76/scripts/yui/assets/skins/sam/skin.css" type="text/css" /><link rel="stylesheet" href="/static/7af60c76/scripts/yui/container/assets/skins/sam/container.css" type="text/css" /><link rel="stylesheet" href="/static/7af60c76/scripts/yui/button/assets/skins/sam/button.css" type="text/css" /><link rel="stylesheet" href="/static/7af60c76/scripts/yui/menu/assets/skins/sam/menu.css" type="text/css" /><meta name="ROBOTS" content="INDEX,NOFOLLOW" /><script src="/static/7af60c76/scripts/yui/cookie/cookie-min.js"></script></head><body id="jenkins" class="yui-skin-sam jenkins-1.609.1" data-version="jenkins-1.609.1"><a href="#skip2content" class="skiplink">Skip to content</a><div id="page-head"><div id="header"><div class="logo"><a id="jenkins-home-link" href="/"><img src="/static/7af60c76/images/headshot.png" alt="title" id="jenkins-head-icon" /><img src="/static/7af60c76/images/title.png" alt="title" width="139" id="jenkins-name-icon" height="34" /></a></div><div class="login"> <a href="/securityRealm/commenceLogin?from=%2Fghprbhook%2F"><b>log in</b></a></div><div class="searchbox hidden-xs"><form method="get" name="search" action="/search/" style="position:relative;" class="no-json"><div id="search-box-minWidth"></div><div id="search-box-sizer"></div><div id="searchform"><input name="q" placeholder="search" id="search-box" class="has-default-text" /> <a href="http://wiki.jenkins-ci.org/display/JENKINS/Search+Box"><img src="/static/7af60c76/images/16x16/help.png" style="width: 16px; height: 16px; " class="icon-help icon-sm" /></a><div id="search-box-completion"></div><script>createSearchBox("/search/");</script></div></form></div></div><div id="breadcrumbBar"><tr id="top-nav"><td id="left-top-nav" colspan="2"><link rel='stylesheet' href='/adjuncts/7af60c76/lib/layout/breadcrumbs.css' type='text/css' /><script src='/adjuncts/7af60c76/lib/layout/breadcrumbs.js' type='text/javascript'></script><div class="top-sticker noedge"><div class="top-sticker-inner"><div id="right-top-nav"></div><ul id="breadcrumbs"><li class="item"><a href="/" class="model-link inside">Jenkins</a></li><li href="/" class="children"></li></ul><div id="breadcrumb-menu-target"></div></div></div></td></tr></div></div><div id="page-body"><div class="row"><div id="side-panel"><div id="side-panel-content"><div class="task"><a href="http://jenkins-ci.org/" class="task-icon-link"><img src="/static/7af60c76/images/24x24/next.png" style="width: 24px; height: 24px; width: 24px; height: 24px; margin: 2px;" class="icon-next icon-md" /></a> <a href="http://jenkins-ci.org/" class="task-link">Jenkins project</a></div><div class="task"><a href="http://issues.jenkins-ci.org/" class="task-icon-link"><img src="/static/7af60c76/images/24x24/gear2.png" style="width: 24px; height: 24px; width: 24px; height: 24px; margin: 2px;" class="icon-gear2 icon-md" /></a> <a href="http://issues.jenkins-ci.org/" class="task-link">Bug tracker</a></div><div class="task"><a href="http://jenkins-ci.org/content/mailing-lists" class="task-icon-link"><img src="/static/7af60c76/images/24x24/search.png" style="width: 24px; height: 24px; width: 24px; height: 24px; margin: 2px;" class="icon-search icon-md" /></a> <a href="http://jenkins-ci.org/content/mailing-lists" class="task-link">Mailing Lists</a></div><div class="task"><a href="https://twitter.com/jenkinsci" class="task-icon-link"><img src="/static/7af60c76/images/24x24/user.png" style="width: 24px; height: 24px; width: 24px; height: 24px; margin: 2px;" class="icon-user icon-md" /></a> <a href="https://twitter.com/jenkinsci" class="task-link">Twitter: @jenkinsci</a></div></div></div><div id="main-panel"><div id="main-panel-content"><a name="skip2content"></a><h1 style="text-align: center"><img src="/static/7af60c76/images/rage.png" width="154" height="179" /><span style="font-size:50px"> Oops!</span></h1><div id="error-description"><p>A problem occurred while processing the request.
        Please check <a href="https://issues.jenkins-ci.org/">our bug tracker</a> to see if a similar problem has already been reported.
        If it is already reported, please vote and put a comment on it to let us gauge the impact of the problem.
        If you think this is a new issue, please file a new issue.
        When you file an issue, make sure to add the entire stack trace, along with the version of Jenkins and relevant plugins.
        <a href="http://jenkins-ci.org/content/mailing-lists">The users list</a> might be also useful in understanding what has happened.</p><h2>Stack trace</h2><pre style="margin:2em; clear:both">javax.servlet.ServletException: java.lang.Error: java.io.FileNotFoundException: {"message":"Not Found","documentation_url":"https://developer.github.com/v3"}
	at org.kohsuke.stapler.Stapler.tryInvoke(Stapler.java:796)
	at org.kohsuke.stapler.Stapler.invoke(Stapler.java:876)
	at org.kohsuke.stapler.MetaClass$13.dispatch(MetaClass.java:411)
	at org.kohsuke.stapler.Stapler.tryInvoke(Stapler.java:746)
	at org.kohsuke.stapler.Stapler.invoke(Stapler.java:876)
	at org.kohsuke.stapler.Stapler.invoke(Stapler.java:649)
	at org.kohsuke.stapler.Stapler.service(Stapler.java:238)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:848)
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:686)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1494)
	at hudson.util.PluginServletFilter$1.doFilter(PluginServletFilter.java:123)
	at hudson.util.PluginServletFilter.doFilter(PluginServletFilter.java:114)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1482)
	at hudson.security.csrf.CrumbFilter.doFilter(CrumbFilter.java:48)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1482)
	at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:84)
	at hudson.security.UnwrapSecurityExceptionFilter.doFilter(UnwrapSecurityExceptionFilter.java:51)
	at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)
	at jenkins.security.ExceptionTranslationFilter.doFilter(ExceptionTranslationFilter.java:117)
	at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)
	at org.acegisecurity.providers.anonymous.AnonymousProcessingFilter.doFilter(AnonymousProcessingFilter.java:125)
	at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)
	at org.acegisecurity.ui.rememberme.RememberMeProcessingFilter.doFilter(RememberMeProcessingFilter.java:135)
	at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)
	at org.acegisecurity.ui.AbstractProcessingFilter.doFilter(AbstractProcessingFilter.java:271)
	at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)
	at jenkins.security.BasicHeaderProcessor.doFilter(BasicHeaderProcessor.java:93)
	at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)
	at org.acegisecurity.context.HttpSessionContextIntegrationFilter.doFilter(HttpSessionContextIntegrationFilter.java:249)
	at hudson.security.HttpSessionContextIntegrationFilter2.doFilter(HttpSessionContextIntegrationFilter2.java:67)
	at hudson.security.ChainedServletFilter$1.doFilter(ChainedServletFilter.java:87)
	at hudson.security.ChainedServletFilter.doFilter(ChainedServletFilter.java:76)
	at hudson.security.HudsonFilter.doFilter(HudsonFilter.java:168)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1482)
	at org.kohsuke.stapler.compression.CompressionFilter.doFilter(CompressionFilter.java:49)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1482)
	at hudson.util.CharacterEncodingFilter.doFilter(CharacterEncodingFilter.java:81)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1482)
	at org.kohsuke.stapler.DiagnosticThreadNameFilter.doFilter(DiagnosticThreadNameFilter.java:30)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1474)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:499)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:137)
	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:533)
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:231)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1086)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:428)
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:193)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1020)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:135)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:116)
	at org.eclipse.jetty.server.Server.handle(Server.java:370)
	at org.eclipse.jetty.server.AbstractHttpConnection.handleRequest(AbstractHttpConnection.java:489)
	at org.eclipse.jetty.server.AbstractHttpConnection.content(AbstractHttpConnection.java:960)
	at org.eclipse.jetty.server.AbstractHttpConnection$RequestHandler.content(AbstractHttpConnection.java:1021)
	at org.eclipse.jetty.http.HttpParser.parseNext(HttpParser.java:865)
	at org.eclipse.jetty.http.HttpParser.parseAvailable(HttpParser.java:240)
	at org.eclipse.jetty.server.AsyncHttpConnection.handle(AsyncHttpConnection.java:82)
	at org.eclipse.jetty.io.nio.SelectChannelEndPoint.handle(SelectChannelEndPoint.java:668)
	at org.eclipse.jetty.io.nio.SelectChannelEndPoint$1.run(SelectChannelEndPoint.java:52)
	at winstone.BoundedExecutorService$1.run(BoundedExecutorService.java:77)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.Error: java.io.FileNotFoundException: {"message":"Not Found","documentation_url":"https://developer.github.com/v3"}
	at org.kohsuke.github.Requester$1.fetch(Requester.java:389)
	at org.kohsuke.github.Requester$1.hasNext(Requester.java:356)
	at org.kohsuke.github.PagedIterator.fetch(PagedIterator.java:44)
	at org.kohsuke.github.PagedIterator.hasNext(PagedIterator.java:32)
	at org.kohsuke.github.PagedIterable.asList(PagedIterable.java:21)
	at org.kohsuke.github.GHIssue.getComments(GHIssue.java:192)
	at org.jenkinsci.plugins.ghprb.GhprbPullRequest.checkComments(GhprbPullRequest.java:347)
	at org.jenkinsci.plugins.ghprb.GhprbPullRequest.check(GhprbPullRequest.java:156)
	at org.jenkinsci.plugins.ghprb.GhprbRepository.onPullRequestHook(GhprbRepository.java:298)
	at org.jenkinsci.plugins.ghprb.GhprbWebHook.handleWebHook(GhprbWebHook.java:61)
	at org.jenkinsci.plugins.ghprb.GhprbRootAction.doIndex(GhprbRootAction.java:89)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.kohsuke.stapler.Function$InstanceFunction.invoke(Function.java:298)
	at org.kohsuke.stapler.Function.bindAndInvoke(Function.java:161)
	at org.kohsuke.stapler.Function.bindAndInvokeAndServeResponse(Function.java:96)
	at org.kohsuke.stapler.MetaClass$2.dispatch(MetaClass.java:165)
	at org.kohsuke.stapler.Stapler.tryInvoke(Stapler.java:746)
	... 62 more
Caused by: java.io.FileNotFoundException: {"message":"Not Found","documentation_url":"https://developer.github.com/v3"}
	at org.kohsuke.github.Requester.handleApiError(Requester.java:494)
	at org.kohsuke.github.Requester$1.fetch(Requester.java:385)
	... 81 more
Caused by: java.io.FileNotFoundException: https://api.github.com/repos/genialis/genesis/issues/599/comments
	at sun.reflect.GeneratedConstructorAccessor87.newInstance(Unknown Source)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:422)
	at sun.net.www.protocol.http.HttpURLConnection$10.run(HttpURLConnection.java:1889)
	at sun.net.www.protocol.http.HttpURLConnection$10.run(HttpURLConnection.java:1884)
	at java.security.AccessController.doPrivileged(Native Method)
	at sun.net.www.protocol.http.HttpURLConnection.getChainedException(HttpURLConnection.java:1883)
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1456)
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1440)
	at sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:254)
	at org.kohsuke.github.Requester.parse(Requester.java:451)
	at org.kohsuke.github.Requester.access$300(Requester.java:63)
	at org.kohsuke.github.Requester$1.fetch(Requester.java:380)
	... 81 more
Caused by: java.io.FileNotFoundException: https://api.github.com/repos/genialis/genesis/issues/599/comments
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1835)
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1440)
	at java.net.HttpURLConnection.getResponseCode(HttpURLConnection.java:480)
	at sun.net.www.protocol.https.HttpsURLConnectionImpl.getResponseCode(HttpsURLConnectionImpl.java:338)
	at org.kohsuke.github.Requester.parse(Requester.java:447)
	... 83 more
</pre></div></div></div></div></div><div id="footer-container" class="hidden-xs"><div id="footer"><span class="page_generated">
          Page generated:
          Jun 24, 2015 4:03:50 PM</span><span class="rest_api"><a href="api/">REST API</a></span><span class="jenkins_ver"><a href="http://jenkins-ci.org/">Jenkins ver. 1.609.1</a></span></div></div></body></html>
```

Jun 24, 2015 3:11:00 PM FINE org.jenkinsci.plugins.ghprb.GhprbTrigger
Use webHooks is set, so not running trigger
Jun 24, 2015 3:11:00 PM FINE org.jenkinsci.plugins.ghprb.GhprbTrigger
Running trigger for zed-cloud-build__PR_AllTest
Jun 24, 2015 3:11:00 PM SEVERE org.jenkinsci.plugins.ghprb.GhprbRepository initGhRepository
Could not retrieve GitHub repository named vmop/zed-cloud-build (Do you have properly set 'GitHub project' field in job configuration?)
java.io.FileNotFoundException: {"message":"Not Found","documentation_url":"https://developer.github.com/v3"}
	at org.kohsuke.github.Requester.handleApiError(Requester.java:494)
	at org.kohsuke.github.Requester._to(Requester.java:245)
	at org.kohsuke.github.Requester.to(Requester.java:191)
	at org.kohsuke.github.GitHub.getRepository(GitHub.java:321)
	at org.jenkinsci.plugins.ghprb.GhprbRepository.initGhRepository(GhprbRepository.java:78)
	at org.jenkinsci.plugins.ghprb.GhprbRepository.check(GhprbRepository.java:88)
	at org.jenkinsci.plugins.ghprb.Ghprb.run(Ghprb.java:134)
	at org.jenkinsci.plugins.ghprb.GhprbTrigger.run(GhprbTrigger.java:208)
	at hudson.triggers.Trigger.checkTriggers(Trigger.java:266)
	at hudson.triggers.Trigger$Cron.doRun(Trigger.java:215)
	at hudson.triggers.SafeTimerTask.run(SafeTimerTask.java:51)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.io.FileNotFoundException: https://api.github.com/repos/vmop/zed-cloud-build
	at sun.reflect.GeneratedConstructorAccessor158.newInstance(Unknown Source)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:422)
	at sun.net.www.protocol.http.HttpURLConnection$10.run(HttpURLConnection.java:1889)
	at sun.net.www.protocol.http.HttpURLConnection$10.run(HttpURLConnection.java:1884)
	at java.security.AccessController.doPrivileged(Native Method)
	at sun.net.www.protocol.http.HttpURLConnection.getChainedException(HttpURLConnection.java:1883)
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1456)
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1440)
	at sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:254)
	at org.kohsuke.github.Requester.parse(Requester.java:451)
	at org.kohsuke.github.Requester._to(Requester.java:224)
	... 16 more
Caused by: java.io.FileNotFoundException: https://api.github.com/repos/vmop/zed-cloud-build
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1835)
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1440)
	at java.net.HttpURLConnection.getResponseCode(HttpURLConnection.java:480)
	at sun.net.www.protocol.https.HttpsURLConnectionImpl.getResponseCode(HttpsURLConnectionImpl.java:338)
	at org.kohsuke.github.Requester.parse(Requester.java:447)
	... 17 more

I'm currently running 1.24.2.

I updated the project configuration, which I guess caused a poll of all of our PRs or something, because again the plugin started failing to update PRs, and it isn't pulling in any new requests, either.

Example stacktrace in the build log:
```
Setting status of b13f2df9195851bdb46f864428368f88aed4c602 to FAILURE with url http://goto.google.com/prkubekins/job/kubernetes-pull-build-test-e2e-gce/1018/ and message: ' 90 tests run, 41 skipped, 1 failed.'
Using context: Jenkins GCE e2e
Could not update commit status of the Pull Request on GitHub.
java.io.InterruptedIOException
	at org.kohsuke.github.RateLimitHandler$1.onError(RateLimitHandler.java:40)
	at org.kohsuke.github.Requester.handleApiError(Requester.java:486)
	at org.kohsuke.github.Requester._to(Requester.java:245)
	at org.kohsuke.github.Requester.to(Requester.java:191)
	at org.kohsuke.github.GHRepository.createCommitStatus(GHRepository.java:780)
	at org.jenkinsci.plugins.ghprb.extensions.status.GhprbSimpleStatus.createCommitStatus(GhprbSimpleStatus.java:179)
	at org.jenkinsci.plugins.ghprb.extensions.status.GhprbSimpleStatus.onBuildComplete(GhprbSimpleStatus.java:160)
	at org.jenkinsci.plugins.ghprb.GhprbBuilds.onCompleted(GhprbBuilds.java:145)
	at org.jenkinsci.plugins.ghprb.GhprbBuildListener.onCompleted(GhprbBuildListener.java:27)
	at org.jenkinsci.plugins.ghprb.GhprbBuildListener.onCompleted(GhprbBuildListener.java:12)
	at hudson.model.listeners.RunListener.fireCompleted(RunListener.java:201)
	at hudson.model.Run.execute(Run.java:1789)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
	at hudson.model.ResourceController.execute(ResourceController.java:98)
	at hudson.model.Executor.run(Executor.java:374)
Caused by: java.io.IOException: Server returned HTTP response code: 403 for URL: https://api.github.com/repos/GoogleCloudPlatform/kubernetes/statuses/b13f2df9195851bdb46f864428368f88aed4c602
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:57)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:526)
	at sun.net.www.protocol.http.HttpURLConnection$6.run(HttpURLConnection.java:1676)
	at sun.net.www.protocol.http.HttpURLConnection$6.run(HttpURLConnection.java:1674)
	at java.security.AccessController.doPrivileged(Native Method)
	at sun.net.www.protocol.http.HttpURLConnection.getChainedException(HttpURLConnection.java:1672)
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1245)
	at sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:254)
	at org.kohsuke.github.Requester.parse(Requester.java:451)
	at org.kohsuke.github.Requester._to(Requester.java:224)
	... 12 more
Caused by: java.io.IOException: Server returned HTTP response code: 403 for URL: https://api.github.com/repos/GoogleCloudPlatform/kubernetes/statuses/b13f2df9195851bdb46f864428368f88aed4c602
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1627)
	at java.net.HttpURLConnection.getResponseCode(HttpURLConnection.java:468)
	at sun.net.www.protocol.https.HttpsURLConnectionImpl.getResponseCode(HttpsURLConnectionImpl.java:338)
	at org.kohsuke.github.Requester.parse(Requester.java:447)
	... 13 more
```

We've now completely exhausted our unauthenticated rate limit:
```
{
  "resources": {
    "core": {
      "limit": 60,
      "remaining": 0,
      "reset": 1435265469
    },
    "search": {
      "limit": 10,
      "remaining": 10,
      "reset": 1435263501
    }
  },
  "rate": {
    "limit": 60,
    "remaining": 0,
    "reset": 1435265469
  }
}
```

Several builds are stuck right now, trying to update the final status. The error message even indicates that the limit was reached:
```
Could not update commit status of the Pull Request on GitHub.
java.io.IOException: {"message":"API rate limit exceeded for 104.154.45.126. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)","documentation_url":"https://developer.github.com/v3/#rate-limiting"}
	at org.kohsuke.github.Requester.handleApiError(Requester.java:496)
	at org.kohsuke.github.Requester._to(Requester.java:245)
	at org.kohsuke.github.Requester.to(Requester.java:191)
	at org.kohsuke.github.GHRepository.createCommitStatus(GHRepository.java:780)
	at org.jenkinsci.plugins.ghprb.extensions.status.GhprbSimpleStatus.createCommitStatus(GhprbSimpleStatus.java:179)
	at org.jenkinsci.plugins.ghprb.extensions.status.GhprbSimpleStatus.onBuildComplete(GhprbSimpleStatus.java:160)
	at org.jenkinsci.plugins.ghprb.GhprbBuilds.onCompleted(GhprbBuilds.java:145)
	at org.jenkinsci.plugins.ghprb.GhprbBuildListener.onCompleted(GhprbBuildListener.java:27)
	at org.jenkinsci.plugins.ghprb.GhprbBuildListener.onCompleted(GhprbBuildListener.java:12)
	at hudson.model.listeners.RunListener.fireCompleted(RunListener.java:201)
	at hudson.model.Run.execute(Run.java:1789)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
	at hudson.model.ResourceController.execute(ResourceController.java:98)
	at hudson.model.Executor.run(Executor.java:374)
Caused by: java.io.IOException: Server returned HTTP response code: 403 for URL: https://api.github.com/repos/GoogleCloudPlatform/kubernetes/statuses/fbd03e5b90aad0cfdf9870e3bd84140626c8f237
	at sun.reflect.GeneratedConstructorAccessor118.newInstance(Unknown Source)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:526)
	at sun.net.www.protocol.http.HttpURLConnection$6.run(HttpURLConnection.java:1676)
	at sun.net.www.protocol.http.HttpURLConnection$6.run(HttpURLConnection.java:1674)
	at java.security.AccessController.doPrivileged(Native Method)
	at sun.net.www.protocol.http.HttpURLConnection.getChainedException(HttpURLConnection.java:1672)
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1245)
	at sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:254)
	at org.kohsuke.github.Requester.parse(Requester.java:451)
	at org.kohsuke.github.Requester._to(Requester.java:224)
	... 12 more
Caused by: java.io.IOException: Server returned HTTP response code: 403 for URL: https://api.github.com/repos/GoogleCloudPlatform/kubernetes/statuses/fbd03e5b90aad0cfdf9870e3bd84140626c8f237
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1627)
	at java.net.HttpURLConnection.getResponseCode(HttpURLConnection.java:468)
	at sun.net.www.protocol.https.HttpsURLConnectionImpl.getResponseCode(HttpsURLConnectionImpl.java:338)
	at org.kohsuke.github.Requester.parse(Requester.java:447)
	... 13 more
```

Maybe something like that, though I'm still a bit confused - it seems to be using our unauth API quota, but it's definitely posting comments as the bot account.

When I converted from GLOBAL to SYSTEM, I guess I replaced the oauth token in the credentials store with something bad (it was autofilled with something - maybe that was junk), which caused some 401s to appear in the logs:
```
Unable to query GitHub for status of PullRequestjava.io.IOException: Server returned HTTP response code: 401 for URL: https://api.github.com/repos/GoogleCloudPlatform/kubernetes/pulls/10246
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:57)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:526)
	at sun.net.www.protocol.http.HttpURLConnection$6.run(HttpURLConnection.java:1676)
	at sun.net.www.protocol.http.HttpURLConnection$6.run(HttpURLConnection.java:1674)
	at java.security.AccessController.doPrivileged(Native Method)
	at sun.net.www.protocol.http.HttpURLConnection.getChainedException(HttpURLConnection.java:1672)
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1245)
	at sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:254)
	at org.kohsuke.github.Requester.parse(Requester.java:451)
	at org.kohsuke.github.Requester._to(Requester.java:224)
	at org.kohsuke.github.Requester.to(Requester.java:198)
	at org.kohsuke.github.GHPullRequest.populate(GHPullRequest.java:196)
	at org.kohsuke.github.GHPullRequest.getMergeable(GHPullRequest.java:169)
	at org.jenkinsci.plugins.ghprb.GhprbBuilds.onStarted(GhprbBuilds.java:72)
	at org.jenkinsci.plugins.ghprb.GhprbBuildListener.onStarted(GhprbBuildListener.java:19)
	at org.jenkinsci.plugins.ghprb.GhprbBuildListener.onStarted(GhprbBuildListener.java:12)
	at hudson.model.listeners.RunListener.fireStarted(RunListener.java:215)
	at hudson.model.Run.execute(Run.java:1740)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
	at hudson.model.ResourceController.execute(ResourceController.java:98)
	at hudson.model.Executor.run(Executor.java:374)
Caused by: java.io.IOException: Server returned HTTP response code: 401 for URL: https://api.github.com/repos/GoogleCloudPlatform/kubernetes/pulls/10246
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1627)
	at java.net.HttpURLConnection.getResponseCode(HttpURLConnection.java:468)
	at sun.net.www.protocol.https.HttpsURLConnectionImpl.getResponseCode(HttpsURLConnectionImpl.java:338)
	at org.kohsuke.github.Requester.parse(Requester.java:447)
	... 12 more
```

Saving the token again, it stopped printing that error, at least, though it still seems stuck.

OK, the build log from a recent run is puzzling, but maybe it's showing what might be part of the issue.

Full log is available [here](https://storage.cloud.google.com/kubernetes-jenkins/pr-logs/f9eaf959d03e1570ddcc6f95c749bff0bfd063ea/kubernetes-pull-build-test-e2e-gce/1085/build-log.txt), corresponding PR is GoogleCloudPlatform/kubernetes/pull/10246.

When the build started, it failed to get PR status:
```
GitHub pull request #10246 of commit f9eaf959d03e1570ddcc6f95c749bff0bfd063ea, no merge conflicts.
Unable to query GitHub for status of PullRequestjava.io.IOException: Server returned HTTP response code: 401 for URL: https://api.github.com/repos/GoogleCloudPlatform/kubernetes/pulls/10246
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:57)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:526)
	at sun.net.www.protocol.http.HttpURLConnection$6.run(HttpURLConnection.java:1676)
	at sun.net.www.protocol.http.HttpURLConnection$6.run(HttpURLConnection.java:1674)
	at java.security.AccessController.doPrivileged(Native Method)
	at sun.net.www.protocol.http.HttpURLConnection.getChainedException(HttpURLConnection.java:1672)
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1245)
	at sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:254)
	at org.kohsuke.github.Requester.parse(Requester.java:451)
	at org.kohsuke.github.Requester._to(Requester.java:224)
	at org.kohsuke.github.Requester.to(Requester.java:198)
	at org.kohsuke.github.GHPullRequest.populate(GHPullRequest.java:196)
	at org.kohsuke.github.GHPullRequest.getMergeable(GHPullRequest.java:169)
	at org.jenkinsci.plugins.ghprb.GhprbBuilds.onStarted(GhprbBuilds.java:72)
	at org.jenkinsci.plugins.ghprb.GhprbBuildListener.onStarted(GhprbBuildListener.java:19)
	at org.jenkinsci.plugins.ghprb.GhprbBuildListener.onStarted(GhprbBuildListener.java:12)
	at hudson.model.listeners.RunListener.fireStarted(RunListener.java:215)
	at hudson.model.Run.execute(Run.java:1740)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
	at hudson.model.ResourceController.execute(ResourceController.java:98)
	at hudson.model.Executor.run(Executor.java:374)
Caused by: java.io.IOException: Server returned HTTP response code: 401 for URL: https://api.github.com/repos/GoogleCloudPlatform/kubernetes/pulls/10246
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1627)
	at java.net.HttpURLConnection.getResponseCode(HttpURLConnection.java:468)
	at sun.net.www.protocol.https.HttpsURLConnectionImpl.getResponseCode(HttpsURLConnectionImpl.java:338)
	at org.kohsuke.github.Requester.parse(Requester.java:447)
	... 12 more
```

But immediately afterwards it (apparently) successfully updated the pending status:
```
Setting status of f9eaf959d03e1570ddcc6f95c749bff0bfd063ea to PENDING with url http://goto.google.com/prkubekins/job/kubernetes-pull-build-test-e2e-gce/1085/ and message: 'Build Started sha1 is merged.'
Using context: Jenkins GCE e2e
```

When the run finished, it commented on the PR and updated the commit status (you can see this in the linked PR):
```
Setting status of f9eaf959d03e1570ddcc6f95c749bff0bfd063ea to FAILURE with url http://goto.google.com/prkubekins/job/kubernetes-pull-build-test-e2e-gce/1085/ and message: ' No test results found.'
Using context: Jenkins GCE e2e
GCE e2e build/test **failed** for commit f9eaf959d03e1570ddcc6f95c749bff0bfd063ea.
* [Build Log](https://storage.cloud.google.com/kubernetes-jenkins/pr-logs/f9eaf959d03e1570ddcc6f95c749bff0bfd063ea/kubernetes-pull-build-test-e2e-gce/1085/build-log.txt)
* [Test Artifacts](https://console.developers.google.com/storage/browser/kubernetes-jenkins/pr-logs/f9eaf959d03e1570ddcc6f95c749bff0bfd063ea/kubernetes-pull-build-test-e2e-gce/1085/_artifacts/)
* [Internal Jenkins Results](http://goto.google.com/prkubekins/job/kubernetes-pull-build-test-e2e-gce//1085)
```

So why did it fail to query PR status, but succeed at updating the commit status and commenting on the commit? When it reads in the list of PRs from `org.jenkinsci.plugins.ghprb.GhprbTrigger.xml`, is it somehow initializing them with an unauthenticated connection or something?

Regarding the "leaked" unauthenticated requests:
I just saw another occurrence of an HTTP 401 while getting PR status:
```
Unable to query GitHub for status of PullRequestjava.io.IOException: Server returned HTTP response code: 401 for URL: https://api.github.com/repos/GoogleCloudPlatform/kubernetes/pulls/10197
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:57)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:526)
	at sun.net.www.protocol.http.HttpURLConnection$6.run(HttpURLConnection.java:1676)
	at sun.net.www.protocol.http.HttpURLConnection$6.run(HttpURLConnection.java:1674)
	at java.security.AccessController.doPrivileged(Native Method)
	at sun.net.www.protocol.http.HttpURLConnection.getChainedException(HttpURLConnection.java:1672)
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1245)
	at sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:254)
	at org.kohsuke.github.Requester.parse(Requester.java:451)
	at org.kohsuke.github.Requester._to(Requester.java:224)
	at org.kohsuke.github.Requester.to(Requester.java:198)
	at org.kohsuke.github.GHPullRequest.populate(GHPullRequest.java:196)
	at org.kohsuke.github.GHPullRequest.getMergeable(GHPullRequest.java:169)
	at org.jenkinsci.plugins.ghprb.GhprbBuilds.onStarted(GhprbBuilds.java:72)
	at org.jenkinsci.plugins.ghprb.GhprbBuildListener.onStarted(GhprbBuildListener.java:19)
	at org.jenkinsci.plugins.ghprb.GhprbBuildListener.onStarted(GhprbBuildListener.java:12)
	at hudson.model.listeners.RunListener.fireStarted(RunListener.java:215)
	at hudson.model.Run.execute(Run.java:1740)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
	at hudson.model.ResourceController.execute(ResourceController.java:98)
	at hudson.model.Executor.run(Executor.java:374)
Caused by: java.io.IOException: Server returned HTTP response code: 401 for URL: https://api.github.com/repos/GoogleCloudPlatform/kubernetes/pulls/10197
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1627)
	at java.net.HttpURLConnection.getResponseCode(HttpURLConnection.java:468)
	at sun.net.www.protocol.https.HttpsURLConnectionImpl.getResponseCode(HttpsURLConnectionImpl.java:338)
	at org.kohsuke.github.Requester.parse(Requester.java:447)
	... 12 more
```

The unauthenticated quota dropped by exactly one request - so it's possible this was retried unauthenticated?

--- a/src/main/java/org/jenkinsci/plugins/ghprb/Ghprb.java
+++ b/src/main/java/org/jenkinsci/plugins/ghprb/Ghprb.java
@@ -8,11 +8,11 @@ import com.cloudbees.plugins.credentials.SystemCredentialsProvider;
 import com.cloudbees.plugins.credentials.common.StandardCredentials;
 import com.cloudbees.plugins.credentials.domains.Domain;
 import com.cloudbees.plugins.credentials.domains.DomainSpecification;
-import com.cloudbees.plugins.credentials.domains.URIRequirementBuilder;
 import com.cloudbees.plugins.credentials.domains.HostnamePortSpecification;
 import com.cloudbees.plugins.credentials.domains.HostnameSpecification;
 import com.cloudbees.plugins.credentials.domains.PathSpecification;
 import com.cloudbees.plugins.credentials.domains.SchemeSpecification;
+import com.cloudbees.plugins.credentials.domains.URIRequirementBuilder;
 import com.cloudbees.plugins.credentials.impl.UsernamePasswordCredentialsImpl;
 import com.coravy.hudson.plugins.github.GithubProjectProperty;
 
@@ -411,7 +411,7 @@ public class Ghprb {
         if (StringUtils.isEmpty(path)) {
             path = "/";
         }
-        specifications.add(new PathSpecification(serverUri.getPath(), null, false));
+        specifications.add(new PathSpecification(path, null, false));
         
         Domain domain = new Domain(serverUri.getHost(), "Auto generated credentials domain", specifications);
         CredentialsStore provider = new SystemCredentialsProvider.StoreImpl();

--- a/src/main/java/org/jenkinsci/plugins/ghprb/GhprbGitHubAuth.java
+++ b/src/main/java/org/jenkinsci/plugins/ghprb/GhprbGitHubAuth.java
@@ -29,7 +29,6 @@ import com.cloudbees.plugins.credentials.common.IdCredentials;
 import com.cloudbees.plugins.credentials.common.StandardCredentials;
 import com.cloudbees.plugins.credentials.common.StandardListBoxModel;
 import com.cloudbees.plugins.credentials.common.StandardUsernamePasswordCredentials;
-import com.cloudbees.plugins.credentials.common.UsernamePasswordCredentials;
 import com.cloudbees.plugins.credentials.domains.DomainRequirement;
 import com.cloudbees.plugins.credentials.domains.HostnamePortRequirement;
 import com.cloudbees.plugins.credentials.domains.HostnameRequirement;
@@ -108,29 +107,43 @@ public class GhprbGitHubAuth extends AbstractDescribableImpl<GhprbGitHubAuth> {
     public String getSecret() {
         return secret;
     }
-    
+
+    private static GitHubBuilder getBuilder(Item context, String serverAPIUrl, String credentialsId) {
+        GitHubBuilder builder = new GitHubBuilder()
+            .withEndpoint(serverAPIUrl)
+            .withConnector(new HttpConnectorWithJenkinsProxy());
+        String contextName = context == null ? "(null)" : context.getFullDisplayName();
+        if (StringUtils.isEmpty(credentialsId)) {
+            logger.log(Level.WARNING, "credentialsId not set for context {0}, using anonymous connection", contextName);
+            return builder;
+        }
+
+        StandardCredentials credentials = Ghprb.lookupCredentials(context, credentialsId, serverAPIUrl);
+        if (credentials == null) {
+            logger.log(Level.SEVERE, "Failed to look up credentials for context {0} using id: {1}",
+                    new Object[] { contextName, credentialsId });
+        } else if (credentials instanceof StandardUsernamePasswordCredentials) {
+            logger.log(Level.FINEST, "Using username/password for context {0}", contextName);
+            StandardUsernamePasswordCredentials upCredentials = (StandardUsernamePasswordCredentials) credentials;
+            builder.withPassword(upCredentials.getUsername(), upCredentials.getPassword().getPlainText());
+        } else if (credentials instanceof StringCredentials) {
+            logger.log(Level.FINEST, "Using OAuth token for context {0}", contextName);
+            StringCredentials tokenCredentials = (StringCredentials) credentials;
+            builder.withOAuthToken(tokenCredentials.getSecret().getPlainText());
+        } else {
+            logger.log(Level.SEVERE, "Unknown credential type for context {0} using id: {1}: {2}",
+                    new Object[] { contextName, credentialsId, credentials.getClass().getName() });
+            return null;
+        }
+        return builder;
+    }
+
     public GitHub getConnection(Item context) throws IOException {
         GitHub gh = null;
-        GitHubBuilder builder = new GitHubBuilder()
-                    .withEndpoint(serverAPIUrl)
-                    .withConnector(new HttpConnectorWithJenkinsProxy());
-        
-        if (!StringUtils.isEmpty(credentialsId)) {
-            StandardCredentials credentials = CredentialsMatchers
-                    .firstOrNull(
-                            CredentialsProvider.lookupCredentials(StandardCredentials.class, context,
-                                    ACL.SYSTEM, URIRequirementBuilder.fromUri(serverAPIUrl).build()),
-                                    CredentialsMatchers.allOf(CredentialsMatchers.withId(credentialsId)));
-            
-            if (credentials instanceof StringCredentials) {
-                String accessToken = ((StringCredentials) credentials).getSecret().getPlainText();
-                builder.withOAuthToken(accessToken);
-            } else if (credentials instanceof UsernamePasswordCredentials){
-                UsernamePasswordCredentials creds = (UsernamePasswordCredentials) credentials;
-                String username = creds.getUsername();
-                String password = creds.getPassword().getPlainText();
-                builder.withPassword(username, password);
-            }
+        GitHubBuilder builder = getBuilder(context, serverAPIUrl, credentialsId);
+        if (builder == null) {
+          logger.log(Level.SEVERE, "Unable to get builder using credentials: {0}", credentialsId);
+          return null;
         }
         try {
             gh = builder.build();
@@ -164,7 +177,7 @@ public class GhprbGitHubAuth extends AbstractDescribableImpl<GhprbGitHubAuth> {
          * @throws URISyntaxException 
          */
         public ListBoxModel doFillCredentialsIdItems(@AncestorInPath Item context, @QueryParameter String serverAPIUrl) throws URISyntaxException {
-            List<DomainRequirement> domainRequirements = getDomainReqs(serverAPIUrl);
+            List<DomainRequirement> domainRequirements = URIRequirementBuilder.fromUri(serverAPIUrl).build();
             
             return new StandardListBoxModel()
                     .withEmptySelection()
@@ -222,24 +235,6 @@ public class GhprbGitHubAuth extends AbstractDescribableImpl<GhprbGitHubAuth> {
             }
         }
         
-        private List<DomainRequirement> getDomainReqs(String serverAPIUrl) throws URISyntaxException {
-            List<DomainRequirement> requirements = new ArrayList<DomainRequirement>(2);
-            
-            URI serverUri = new URI(serverAPIUrl);
-            
-            if (serverUri.getPort() > 0) {
-                requirements.add(new HostnamePortRequirement(serverUri.getHost(), serverUri.getPort()));
-            } else {
-                requirements.add(new HostnameRequirement(serverUri.getHost()));
-            }
-            
-            requirements.add(new SchemeRequirement(serverUri.getScheme()));
-            if (!StringUtils.isEmpty(serverUri.getPath())) {
-                requirements.add(new PathRequirement(serverUri.getPath()));
-            }
-            return requirements;
-        }
-
         public FormValidation doCheckServerAPIUrl(@QueryParameter String value) {
             if ("https://api.github.com".equals(value)) {
                 return FormValidation.ok();
@@ -255,9 +250,9 @@ public class GhprbGitHubAuth extends AbstractDescribableImpl<GhprbGitHubAuth> {
                 @QueryParameter("credentialsId") final String credentialsId,
                 @QueryParameter("repo") final String repo) {
             try {
-                GitHubBuilder builder = getBuilder(serverAPIUrl, credentialsId);
+                GitHubBuilder builder = getBuilder(null, serverAPIUrl, credentialsId);
                 if (builder == null) {
-                    return FormValidation.error("No credentials ID provided!!");
+                    return FormValidation.error("Unable to look up GitHub credentials using ID: " + credentialsId + "!!");
                 }
                 GitHub gh = builder.build();
                 GHRepository repository = gh.getRepository(repo);
@@ -281,32 +276,13 @@ public class GhprbGitHubAuth extends AbstractDescribableImpl<GhprbGitHubAuth> {
             }
         }
         
-        private GitHubBuilder getBuilder(String serverAPIUrl, String credentialsId) {
-            GitHubBuilder builder = new GitHubBuilder()
-                .withEndpoint(serverAPIUrl)
-                .withConnector(new HttpConnectorWithJenkinsProxy());
-
-            StandardCredentials credentials = Ghprb.lookupCredentials(null, credentialsId, serverAPIUrl);
-            if (credentials instanceof StandardUsernamePasswordCredentials) {
-                StandardUsernamePasswordCredentials upCredentials = (StandardUsernamePasswordCredentials) credentials;
-                builder.withPassword(upCredentials.getUsername(), upCredentials.getPassword().getPlainText());
-                
-            } else if (credentials instanceof StringCredentials) {
-                StringCredentials tokenCredentials = (StringCredentials) credentials;
-                builder.withOAuthToken(tokenCredentials.getSecret().getPlainText());
-            } else {
-                return null;
-            }
-            return builder;
-        }
-
         public FormValidation doTestGithubAccess(
                 @QueryParameter("serverAPIUrl") final String serverAPIUrl, 
                 @QueryParameter("credentialsId") final String credentialsId) {
             try {
-                GitHubBuilder builder = getBuilder(serverAPIUrl, credentialsId);
+                GitHubBuilder builder = getBuilder(null, serverAPIUrl, credentialsId);
                 if (builder == null) {
-                    return FormValidation.error("No credentials ID provided!!");
+                    return FormValidation.error("Unable to look up GitHub credentials using ID: " + credentialsId + "!!");
                 }
                 GitHub gh = builder.build();
                 GHMyself me = gh.getMyself();

--- a/src/main/java/org/jenkinsci/plugins/ghprb/GhprbTrigger.java
+++ b/src/main/java/org/jenkinsci/plugins/ghprb/GhprbTrigger.java
@@ -193,7 +193,7 @@ public class GhprbTrigger extends GhprbTriggerBackwardsCompatible {
         }
 
         if ((helper != null && helper.isProjectDisabled()) || (_project != null && _project.isDisabled())) {
-            logger.log(Level.FINE, "Project is disabled, ignoring trigger run call");
+            logger.log(Level.FINE, "Project is disabled, ignoring trigger run call for job {0}", this.project);
             return;
         }
         

